app/controllers/admin/cases_controller.rb

```diff
@@ -97,7 +87,43 @@ def update
         @case.publisher.status_updates.create(status: PublisherStatusUpdate::ACTIVE, publisher_note: note)
       end
 
-      redirect_to admin_cases_path
+      return redirect_to [:admin, @case], flash: { notice: "Case has been moved back to in progress"} if params[:status] == Case::IN_PROGRESS
+
+      next_case = Case.where(assignee: current_user, status: Case::IN_PROGRESS).take
+      if next_case
+        redirect_to [:admin, next_case], flash: { notice: "Previous case #{@case.number} has been marked #{params[:status]}"}
+      else
+        redirect_to admin_cases_path
+      end
+    end
+
+    private
+
+    def parse_search(search_case, queries)
+      statuses = queries.select { |x| x.include?("status") }.map { |x| x.split(':').last&.gsub(/[^0-9a-z_]/i, '') }
+      assigned = queries.select { |x| x.include?("assigned") }.map { |x| x.split(':').last&.gsub(/[^0-9a-z_]/i, '') }
+
+      first_status = statuses[0]
+      if first_status.present?
+        value = first_status
+        search_case = search_case.where(status: value)
+
+        statuses[1..-1].each do |value|
+          search_case = search_case.or(Case.where('status LIKE ?', "%#{value}%"))
+        end
+      end
+
+      first_assigned = assigned[0]
+      if first_assigned.present?
+        value = first_assigned
+        search_case = search_case.joins(:assignee).where('publishers.email LIKE ?', "%#{value}%")
```

Comment:
Would it be okay to use `publishers.email LIKE ?, "#{value}%` instead to make use of the index? Otherwise I'd prefer we'd read the read replica.

Same request applies to line 122

Reply:
Sure thing! https://github.com/brave-intl/publishers/pull/2627/commits/4c5a3f976848852c38f029bd429474a475fb26e3